db/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+pub use sqlx;
+use sqlx::postgres::PgPoolOptions;
+pub use sqlx::PgPool;
+
+#[tracing::instrument(err)]
+pub async fn setup_db_pool() -> color_eyre::Result<PgPool> {
```

Comment:
This should only needed to be called once in the CLI to setup the Database, and then you can pass around (and clone if needed!) the `PgPool` from there!

But this uses the `DATABASE_URL` env var to connect to the database, and then it runs migrations. The AdvisoryLock stuff is to make sure that only 1 server at a time is trying to run migrations against the DB. Not needed now for local dev and stuff but might be when we deploy to prod so good to setup now!